src/main/java/com/iridium/iridiumskyblock/listeners/BucketListener.java

```diff
@@ -51,48 +50,41 @@ public void onBucketEvent(PlayerBucketEvent event) {
 
     @EventHandler
     public void onClick(PlayerInteractEvent event) {
-        // I tested this on 1.17.1 so if it doesnt work on any version lower, they can suck it kekW
-
-        final Block clickedBlock = event.getClickedBlock();
-        if (clickedBlock == null)
-            return;
-
-        if (event.getHand() != EquipmentSlot.HAND)
-            return;
-
-        final ItemStack handItem = event.getItem();
-        if (handItem == null)
-            return;
-
-        if (handItem.getType() != Material.BUCKET)
-            return;
-
-        if (clickedBlock.getType() != Material.OBSIDIAN)
-            return;
-
-        if (!IridiumSkyblock.getInstance().getConfiguration().obsidianBucket)
-            return;
-
-        if (event.getPlayer().isSneaking())
-            return;
-
-        final IslandManager manager = IridiumSkyblock.getInstance().getIslandManager();
-
-        final Optional<Island> islandOptional = manager.getIslandViaLocation(event.getClickedBlock().getLocation());
-        if (!islandOptional.isPresent())
-            return;
-
-        final User user = IridiumSkyblock.getInstance().getUserManager().getUser(event.getPlayer());
+        Player player = event.getPlayer();
+        ItemStack itemInHand = player.getInventory().getItemInHand();
 
-        if (!manager.getIslandPermission(islandOptional.get(), user, PermissionType.BUCKET))
+        if (!(IridiumSkyblock.getInstance().getConfiguration().obsidianBucket
+                && event.getAction().equals(Action.RIGHT_CLICK_BLOCK)
+                && event.getClickedBlock().getType().equals(Material.OBSIDIAN)
+                && itemInHand.getType().equals(Material.BUCKET))) {
             return;
+        }
 
-        handItem.setAmount(handItem.getAmount() - 1);
-        event.getClickedBlock().setType(Material.AIR);
-        // Change item in hand a tick later to prevent lava duplication
-        Bukkit.getScheduler().runTaskLater(IridiumSkyblock.getInstance(), () -> event.getPlayer().getInventory().addItem(new ItemStack(Material.LAVA_BUCKET)), 1);
+        Location location = event.getClickedBlock().getLocation();
+        Optional<Island> island = IridiumSkyblock.getInstance().getIslandManager().getIslandViaLocation(location);
+        User user = IridiumSkyblock.getInstance().getUserManager().getUser(player);
 
-        event.setCancelled(true);
+        if (island.isPresent()) {
+            if (IridiumSkyblock.getInstance().getIslandManager().getIslandPermission(island.get(), user, PermissionType.BUCKET)) {
+                event.getClickedBlock().setType(Material.AIR);
+                if (itemInHand.getAmount() > 1) {
+                    itemInHand.setAmount(itemInHand.getAmount() - 1);
+                    Bukkit.getScheduler().runTaskLater(IridiumSkyblock.getInstance(), () -> {
```

Comment:
I dont get why we are running this a tick later, can somone explain what this lava duplication thing is?

Reply:
i just edited from Ori's code lol